{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import collections\n",
    "import matplotlib.pyplot as plt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Condition to test: A22.0P5UM\n"
     ]
    }
   ],
   "source": [
    "condition = input('Condition to test: ')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [],
   "source": [
    "all_phenotypes = pd.read_csv('/home/claudio/Comp_Genetics/results/bugwas_phenotypes.tsv', sep = '\\t')\n",
    "genotypes = pd.read_csv('/home/claudio/Comp_Genetics/raw_data/genotypes_noref_formatted.tsv', sep = '\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [],
   "source": [
    "subset_phenotypes = all_phenotypes[all_phenotypes['pheno'].str.contains(condition)]\n",
    "genotypes_1 = genotypes.rename(columns={\"Unnamed: 0\":\"ps\"})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [],
   "source": [
    "ini_samples_genotypes = list(genotypes_1.columns.values[2:])\n",
    "ini_samples_phenotypes = list(subset_phenotypes.iloc[:,0])\n",
    "\n",
    "# Removing the reference sample from the phenotypes (aka NT12001)\n",
    "ini_samples_phenotypes = [sample for sample in ini_samples_phenotypes if sample != 'NT12001']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Which are the samples that do not match\n",
    "nomatch_samples = [sample for sample in ini_samples_phenotypes if sample not in ini_samples_genotypes]\n",
    "nomatch_samples = list(dict.fromkeys(nomatch_samples))\n",
    "    \n",
    "# Get rid of duplicates and nomatching samples on the phenotypes\n",
    "phenotypes_1 = subset_phenotypes.drop_duplicates(subset=['ID'])\n",
    "phenotypes_2 = phenotypes_1[~phenotypes_1['ID'].isin(nomatch_samples) & phenotypes_1['ID'].isin(ini_samples_phenotypes)] ## ~ = negation\n",
    "    \n",
    "# Get sample list to filter\n",
    "samples_phenotypes = list(phenotypes_2.iloc[:,0])\n",
    "samples_phenotypes.insert(0,'ps')\n",
    "    \n",
    "# Subset the genotype dataframe with the phenotype samples\n",
    "subset_genotypes = genotypes_1[samples_phenotypes]\n",
    "    \n",
    "# Remove invariant sites\n",
    "invariant_sites = []\n",
    "for i, k in subset_genotypes.iterrows():\n",
    "    if len(set(k[1:]))== 1:\n",
    "        invariant_sites.append(False)\n",
    "    else:\n",
    "        invariant_sites.append(True)\n",
    "subset_genotypes = subset_genotypes[invariant_sites]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/claudio/Comp_Genetics/tassel-5-standalone/lib/ahocorasick-0.2.4.jar:/home/claudio/Comp_Genetics/tassel-5-standalone/lib/kotlinx-coroutines-core-1.3.0.jar:/home/claudio/Comp_Genetics/tassel-5-standalone/lib/kotlin-stdlib-1.3.50.jar:/home/claudio/Comp_Genetics/tassel-5-standalone/lib/jfreechart-1.0.19.jar:/home/claudio/Comp_Genetics/tassel-5-standalone/lib/slf4j-api-1.7.10.jar:/home/claudio/Comp_Genetics/tassel-5-standalone/lib/biojava-core-4.0.0.jar:/home/claudio/Comp_Genetics/tassel-5-standalone/lib/itextpdf-5.1.0.jar:/home/claudio/Comp_Genetics/tassel-5-standalone/lib/log4j-1.2.13.jar:/home/claudio/Comp_Genetics/tassel-5-standalone/lib/gs-core-1.3.jar:/home/claudio/Comp_Genetics/tassel-5-standalone/lib/snappy-java-1.1.1.6.jar:/home/claudio/Comp_Genetics/tassel-5-standalone/lib/fastutil-8.2.2.jar:/home/claudio/Comp_Genetics/tassel-5-standalone/lib/slf4j-simple-1.7.10.jar:/home/claudio/Comp_Genetics/tassel-5-standalone/lib/scala-library-2.10.1.jar:/home/claudio/Comp_Genetics/tassel-5-standalone/lib/commons-codec-1.10.jar:/home/claudio/Comp_Genetics/tassel-5-standalone/lib/htsjdk-2.19.0.jar:/home/claudio/Comp_Genetics/tassel-5-standalone/lib/javax.json-1.0.4.jar:/home/claudio/Comp_Genetics/tassel-5-standalone/lib/biojava-phylo-4.0.0.jar:/home/claudio/Comp_Genetics/tassel-5-standalone/lib/trove-3.0.3.jar:/home/claudio/Comp_Genetics/tassel-5-standalone/lib/colt-1.2.0.jar:/home/claudio/Comp_Genetics/tassel-5-standalone/lib/junit-4.10.jar:/home/claudio/Comp_Genetics/tassel-5-standalone/lib/forester-1.038.jar:/home/claudio/Comp_Genetics/tassel-5-standalone/lib/ejml-0.23.jar:/home/claudio/Comp_Genetics/tassel-5-standalone/lib/gs-ui-1.3.jar:/home/claudio/Comp_Genetics/tassel-5-standalone/lib/jhdf5-14.12.5.jar:/home/claudio/Comp_Genetics/tassel-5-standalone/lib/sqlite-jdbc-3.8.5-pre1.jar:/home/claudio/Comp_Genetics/tassel-5-standalone/lib/jfreesvg-3.2.jar:/home/claudio/Comp_Genetics/tassel-5-standalone/lib/phg.jar:/home/claudio/Comp_Genetics/tassel-5-standalone/lib/jcommon-1.0.23.jar:/home/claudio/Comp_Genetics/tassel-5-standalone/lib/biojava-alignment-4.0.0.jar:/home/claudio/Comp_Genetics/tassel-5-standalone/lib/guava-22.0.jar:/home/claudio/Comp_Genetics/tassel-5-standalone/lib/postgresql-9.4-1201.jdbc41.jar:/home/claudio/Comp_Genetics/tassel-5-standalone/lib/commons-math3-3.4.1.jar:/home/claudio/Comp_Genetics/tassel-5-standalone/lib/json-simple-1.1.1.jar:/home/claudio/Comp_Genetics/tassel-5-standalone/lib/mail-1.4.jar:/home/claudio/Comp_Genetics/tassel-5-standalone/sTASSEL.jar\n",
      "Memory Settings: -Xms512m -Xmx10g\n",
      "Tassel Pipeline Arguments: -importGuess /home/claudio/Comp_Genetics/input/snps.vcf.gz -CreateTreePlugin -clusteringMethod Neighbor_Joining -endPlugin -export -exportType SqrMatrix\n",
      "[main] INFO net.maizegenetics.tassel.TasselLogging - Tassel Version: 5.2.55  Date: September 10, 2019\n",
      "[main] INFO net.maizegenetics.tassel.TasselLogging - Max Available Memory Reported by JVM: 9102 MB\n",
      "[main] INFO net.maizegenetics.tassel.TasselLogging - Java Version: 1.8.0_191\n",
      "[main] INFO net.maizegenetics.tassel.TasselLogging - OS: Linux\n",
      "[main] INFO net.maizegenetics.tassel.TasselLogging - Number of Processors: 32\n",
      "[main] INFO net.maizegenetics.pipeline.TasselPipeline - Tassel Pipeline Arguments: [-fork1, -importGuess, /home/claudio/Comp_Genetics/input/snps.vcf.gz, -CreateTreePlugin, -clusteringMethod, Neighbor_Joining, -endPlugin, -export, -exportType, SqrMatrix, -runfork1]\n",
      "net.maizegenetics.analysis.data.FileLoadPlugin\n",
      "   net.maizegenetics.analysis.tree.CreateTreePlugin\n",
      "      net.maizegenetics.analysis.data.ExportMultiplePlugin\n",
      "[pool-1-thread-1] INFO net.maizegenetics.plugindef.AbstractPlugin - Starting net.maizegenetics.analysis.data.FileLoadPlugin: time: Sep 16, 2019 18:08:31\n",
      "[pool-1-thread-1] INFO net.maizegenetics.plugindef.AbstractPlugin - \n",
      "FileLoadPlugin Parameters\n",
      "format: Make Best Guess\n",
      "sortPositions: false\n",
      "keepDepth: true\n",
      "\n",
      "[pool-1-thread-1] INFO net.maizegenetics.analysis.data.FileLoadPlugin - Start Loading File: /home/claudio/Comp_Genetics/input/snps.vcf.gz time: Sep 16, 2019 18:08:31\n",
      "[pool-1-thread-1] INFO net.maizegenetics.analysis.data.FileLoadPlugin - guessAtUnknowns: type: VCF\n",
      "[pool-1-thread-1] INFO net.maizegenetics.pipeline.TasselPipeline - net.maizegenetics.analysis.data.FileLoadPlugin: time: Sep 16, 2019 18:08:36: progress: 0%\n",
      "[pool-1-thread-1] INFO net.maizegenetics.pipeline.TasselPipeline - net.maizegenetics.analysis.data.FileLoadPlugin: time: Sep 16, 2019 18:08:38: progress: 10%\n",
      "[pool-1-thread-1] INFO net.maizegenetics.pipeline.TasselPipeline - net.maizegenetics.analysis.data.FileLoadPlugin: time: Sep 16, 2019 18:08:38: progress: 20%\n",
      "[pool-1-thread-1] INFO net.maizegenetics.pipeline.TasselPipeline - net.maizegenetics.analysis.data.FileLoadPlugin: time: Sep 16, 2019 18:08:39: progress: 30%\n",
      "[pool-1-thread-1] INFO net.maizegenetics.pipeline.TasselPipeline - net.maizegenetics.analysis.data.FileLoadPlugin: time: Sep 16, 2019 18:08:39: progress: 40%\n",
      "[pool-1-thread-1] INFO net.maizegenetics.pipeline.TasselPipeline - net.maizegenetics.analysis.data.FileLoadPlugin: time: Sep 16, 2019 18:08:39: progress: 50%\n",
      "[pool-1-thread-1] INFO net.maizegenetics.pipeline.TasselPipeline - net.maizegenetics.analysis.data.FileLoadPlugin: time: Sep 16, 2019 18:08:39: progress: 60%\n",
      "[pool-1-thread-1] INFO net.maizegenetics.pipeline.TasselPipeline - net.maizegenetics.analysis.data.FileLoadPlugin: time: Sep 16, 2019 18:08:40: progress: 70%\n",
      "[pool-1-thread-1] INFO net.maizegenetics.pipeline.TasselPipeline - net.maizegenetics.analysis.data.FileLoadPlugin: time: Sep 16, 2019 18:08:40: progress: 80%\n",
      "[pool-1-thread-1] INFO net.maizegenetics.pipeline.TasselPipeline - net.maizegenetics.analysis.data.FileLoadPlugin: time: Sep 16, 2019 18:08:40: progress: 90%\n",
      "[pool-1-thread-1] INFO net.maizegenetics.pipeline.TasselPipeline - net.maizegenetics.analysis.data.FileLoadPlugin: time: Sep 16, 2019 18:08:41: progress: 100%\n",
      "[pool-1-thread-1] INFO net.maizegenetics.plugindef.AbstractPlugin - net.maizegenetics.analysis.data.FileLoadPlugin  Citation: Bradbury PJ, Zhang Z, Kroon DE, Casstevens TM, Ramdoss Y, Buckler ES. (2007) TASSEL: Software for association mapping of complex traits in diverse samples. Bioinformatics 23:2633-2635.\n",
      "BuilderFromVCF data timing 12.3589s \n",
      "[pool-1-thread-1] INFO net.maizegenetics.analysis.data.FileLoadPlugin - Finished Loading File: /home/claudio/Comp_Genetics/input/snps.vcf.gz time: Sep 16, 2019 18:08:43\n",
      "Genotype Table Name: snps\n",
      "Number of Taxa: 695\n",
      "Number of Sites: 648562\n",
      "Sites x Taxa: 450750590\n",
      "Chromosomes...\n",
      "NC_000913.3: start site: 0 (64) last site: 648561 (4641622) total: 648562\n",
      "\n",
      "[pool-1-thread-1] INFO net.maizegenetics.plugindef.AbstractPlugin - Finished net.maizegenetics.analysis.data.FileLoadPlugin: time: Sep 16, 2019 18:08:43\n",
      "[pool-1-thread-1] INFO net.maizegenetics.plugindef.AbstractPlugin - Starting net.maizegenetics.analysis.tree.CreateTreePlugin: time: Sep 16, 2019 18:08:43\n",
      "[pool-1-thread-1] INFO net.maizegenetics.plugindef.AbstractPlugin - \n",
      "CreateTreePlugin Parameters\n",
      "clusteringMethod: Neighbor_Joining\n",
      "saveDistanceMatrix: true\n",
      "\n",
      "[ForkJoinPool.commonPool-worker-7] INFO net.maizegenetics.pipeline.TasselPipeline - net.maizegenetics.analysis.tree.CreateTreePlugin: time: Sep 16, 2019 18:08:44: progress: 0%\n",
      "[ForkJoinPool.commonPool-worker-15] INFO net.maizegenetics.pipeline.TasselPipeline - net.maizegenetics.analysis.tree.CreateTreePlugin: time: Sep 16, 2019 18:08:46: progress: 12%\n",
      "[ForkJoinPool.commonPool-worker-16] INFO net.maizegenetics.pipeline.TasselPipeline - net.maizegenetics.analysis.tree.CreateTreePlugin: time: Sep 16, 2019 18:08:46: progress: 22%\n",
      "[ForkJoinPool.commonPool-worker-12] INFO net.maizegenetics.pipeline.TasselPipeline - net.maizegenetics.analysis.tree.CreateTreePlugin: time: Sep 16, 2019 18:08:46: progress: 32%\n",
      "[ForkJoinPool.commonPool-worker-27] INFO net.maizegenetics.pipeline.TasselPipeline - net.maizegenetics.analysis.tree.CreateTreePlugin: time: Sep 16, 2019 18:08:46: progress: 41%\n",
      "[ForkJoinPool.commonPool-worker-20] INFO net.maizegenetics.pipeline.TasselPipeline - net.maizegenetics.analysis.tree.CreateTreePlugin: time: Sep 16, 2019 18:08:46: progress: 51%\n",
      "[ForkJoinPool.commonPool-worker-18] INFO net.maizegenetics.pipeline.TasselPipeline - net.maizegenetics.analysis.tree.CreateTreePlugin: time: Sep 16, 2019 18:08:46: progress: 61%\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[ForkJoinPool.commonPool-worker-3] INFO net.maizegenetics.pipeline.TasselPipeline - net.maizegenetics.analysis.tree.CreateTreePlugin: time: Sep 16, 2019 18:08:46: progress: 70%\n",
      "[ForkJoinPool.commonPool-worker-23] INFO net.maizegenetics.pipeline.TasselPipeline - net.maizegenetics.analysis.tree.CreateTreePlugin: time: Sep 16, 2019 18:08:46: progress: 80%\n",
      "[ForkJoinPool.commonPool-worker-25] INFO net.maizegenetics.pipeline.TasselPipeline - net.maizegenetics.analysis.tree.CreateTreePlugin: time: Sep 16, 2019 18:08:46: progress: 90%\n",
      "[ForkJoinPool.commonPool-worker-21] INFO net.maizegenetics.pipeline.TasselPipeline - net.maizegenetics.analysis.tree.CreateTreePlugin: time: Sep 16, 2019 18:08:46: progress: 100%\n",
      "[pool-1-thread-1] INFO net.maizegenetics.analysis.distance.IBSDistanceMatrix3Alleles - IBSDistanceMatrix3Alleles: computeHetBitDistances time = 2 seconds\n",
      "[pool-1-thread-1] INFO net.maizegenetics.plugindef.AbstractPlugin - Finished net.maizegenetics.analysis.tree.CreateTreePlugin: time: Sep 16, 2019 18:08:46\n",
      "[pool-1-thread-1] INFO net.maizegenetics.plugindef.AbstractPlugin - Starting net.maizegenetics.analysis.data.ExportPlugin: time: Sep 16, 2019 18:08:46\n",
      "[pool-1-thread-1] INFO net.maizegenetics.plugindef.AbstractPlugin - \n",
      "ExportPlugin Parameters\n",
      "saveAs: Tree:snps\n",
      "format: Square Matrix\n",
      "keepDepth: true\n",
      "includeTaxaAnnotations: true\n",
      "\n",
      "[pool-1-thread-1] INFO net.maizegenetics.analysis.data.ExportPlugin - performFunction: wrote dataset: Tree:snps to file: Tree:snps.txt\n",
      "[pool-1-thread-1] INFO net.maizegenetics.plugindef.AbstractPlugin - Finished net.maizegenetics.analysis.data.ExportPlugin: time: Sep 16, 2019 18:08:46\n",
      "[pool-1-thread-1] INFO net.maizegenetics.plugindef.AbstractPlugin - Starting net.maizegenetics.analysis.data.ExportPlugin: time: Sep 16, 2019 18:08:46\n",
      "[pool-1-thread-1] INFO net.maizegenetics.plugindef.AbstractPlugin - \n",
      "ExportPlugin Parameters\n",
      "saveAs: Matrix:snps\n",
      "format: Square Matrix\n",
      "keepDepth: true\n",
      "includeTaxaAnnotations: true\n",
      "\n",
      "[pool-1-thread-1] INFO net.maizegenetics.analysis.data.ExportPlugin - performFunction: wrote dataset: Matrix:snps to file: Matrix:snps.txt\n",
      "[pool-1-thread-1] INFO net.maizegenetics.plugindef.AbstractPlugin - Finished net.maizegenetics.analysis.data.ExportPlugin: time: Sep 16, 2019 18:08:46\n"
     ]
    }
   ],
   "source": [
    "!/home/claudio/Comp_Genetics/tassel-5-standalone/run_pipeline.pl -Xms512m -Xmx10g -importGuess /home/claudio/Comp_Genetics/input/snps.vcf.gz -CreateTreePlugin -clusteringMethod Neighbor_Joining -endPlugin -export -exportType SqrMatrix\n",
    "!mv /home/claudio/Comp_Genetics/tassel-5-standalone/Matrix:snps.txt /home/claudio/Comp_Genetics/input/Distance_matrix.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {},
   "outputs": [],
   "source": [
    "distance_matrix = pd.read_csv('/home/claudio/Comp_Genetics/input/Distance_matrix.tsv', sep = '\\t', header = None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>9</th>\n",
       "      <th>...</th>\n",
       "      <th>686</th>\n",
       "      <th>687</th>\n",
       "      <th>688</th>\n",
       "      <th>689</th>\n",
       "      <th>690</th>\n",
       "      <th>691</th>\n",
       "      <th>692</th>\n",
       "      <th>693</th>\n",
       "      <th>694</th>\n",
       "      <th>695</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>NT12219</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.016369</td>\n",
       "      <td>0.009429</td>\n",
       "      <td>0.010343</td>\n",
       "      <td>0.058500</td>\n",
       "      <td>0.011262</td>\n",
       "      <td>0.000680</td>\n",
       "      <td>0.070663</td>\n",
       "      <td>0.058651</td>\n",
       "      <td>...</td>\n",
       "      <td>0.057861</td>\n",
       "      <td>0.055464</td>\n",
       "      <td>0.041451</td>\n",
       "      <td>0.000519</td>\n",
       "      <td>0.057934</td>\n",
       "      <td>0.125000</td>\n",
       "      <td>0.010252</td>\n",
       "      <td>0.010522</td>\n",
       "      <td>0.010153</td>\n",
       "      <td>0.059328</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>NT12588</td>\n",
       "      <td>0.016369</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.014328</td>\n",
       "      <td>0.015217</td>\n",
       "      <td>0.056194</td>\n",
       "      <td>0.017083</td>\n",
       "      <td>0.013521</td>\n",
       "      <td>0.076053</td>\n",
       "      <td>0.056446</td>\n",
       "      <td>...</td>\n",
       "      <td>0.056339</td>\n",
       "      <td>0.065170</td>\n",
       "      <td>0.038674</td>\n",
       "      <td>0.013868</td>\n",
       "      <td>0.056473</td>\n",
       "      <td>0.083333</td>\n",
       "      <td>0.016821</td>\n",
       "      <td>0.017205</td>\n",
       "      <td>0.015589</td>\n",
       "      <td>0.059486</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>NT12359</td>\n",
       "      <td>0.009429</td>\n",
       "      <td>0.014328</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.009740</td>\n",
       "      <td>0.053117</td>\n",
       "      <td>0.010871</td>\n",
       "      <td>0.009328</td>\n",
       "      <td>0.081058</td>\n",
       "      <td>0.058294</td>\n",
       "      <td>...</td>\n",
       "      <td>0.053201</td>\n",
       "      <td>0.063365</td>\n",
       "      <td>0.032258</td>\n",
       "      <td>0.009290</td>\n",
       "      <td>0.053271</td>\n",
       "      <td>0.142857</td>\n",
       "      <td>0.010199</td>\n",
       "      <td>0.010225</td>\n",
       "      <td>0.009506</td>\n",
       "      <td>0.058116</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>NT12611</td>\n",
       "      <td>0.010343</td>\n",
       "      <td>0.015217</td>\n",
       "      <td>0.009740</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.055359</td>\n",
       "      <td>0.009888</td>\n",
       "      <td>0.009260</td>\n",
       "      <td>0.080031</td>\n",
       "      <td>0.057796</td>\n",
       "      <td>...</td>\n",
       "      <td>0.055181</td>\n",
       "      <td>0.062853</td>\n",
       "      <td>0.038462</td>\n",
       "      <td>0.009600</td>\n",
       "      <td>0.055303</td>\n",
       "      <td>0.095238</td>\n",
       "      <td>0.010101</td>\n",
       "      <td>0.001640</td>\n",
       "      <td>0.000225</td>\n",
       "      <td>0.056974</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>NT12856</td>\n",
       "      <td>0.058500</td>\n",
       "      <td>0.056194</td>\n",
       "      <td>0.053117</td>\n",
       "      <td>0.055359</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.061167</td>\n",
       "      <td>0.053473</td>\n",
       "      <td>0.060109</td>\n",
       "      <td>0.021639</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.049782</td>\n",
       "      <td>0.003058</td>\n",
       "      <td>0.053902</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.117647</td>\n",
       "      <td>0.059919</td>\n",
       "      <td>0.062591</td>\n",
       "      <td>0.058593</td>\n",
       "      <td>0.020034</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>690</td>\n",
       "      <td>NT12184</td>\n",
       "      <td>0.125000</td>\n",
       "      <td>0.083333</td>\n",
       "      <td>0.142857</td>\n",
       "      <td>0.095238</td>\n",
       "      <td>0.117647</td>\n",
       "      <td>0.050000</td>\n",
       "      <td>0.083333</td>\n",
       "      <td>0.066667</td>\n",
       "      <td>0.111111</td>\n",
       "      <td>...</td>\n",
       "      <td>0.117647</td>\n",
       "      <td>0.125000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.090909</td>\n",
       "      <td>0.117647</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.150000</td>\n",
       "      <td>0.090909</td>\n",
       "      <td>0.047619</td>\n",
       "      <td>0.105263</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>691</td>\n",
       "      <td>NT12612</td>\n",
       "      <td>0.010252</td>\n",
       "      <td>0.016821</td>\n",
       "      <td>0.010199</td>\n",
       "      <td>0.010101</td>\n",
       "      <td>0.059919</td>\n",
       "      <td>0.011269</td>\n",
       "      <td>0.009738</td>\n",
       "      <td>0.070707</td>\n",
       "      <td>0.059038</td>\n",
       "      <td>...</td>\n",
       "      <td>0.059298</td>\n",
       "      <td>0.052690</td>\n",
       "      <td>0.032967</td>\n",
       "      <td>0.010199</td>\n",
       "      <td>0.059373</td>\n",
       "      <td>0.150000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.010512</td>\n",
       "      <td>0.009823</td>\n",
       "      <td>0.057306</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>692</td>\n",
       "      <td>NT12452</td>\n",
       "      <td>0.010522</td>\n",
       "      <td>0.017205</td>\n",
       "      <td>0.010225</td>\n",
       "      <td>0.001640</td>\n",
       "      <td>0.062591</td>\n",
       "      <td>0.010555</td>\n",
       "      <td>0.009252</td>\n",
       "      <td>0.089769</td>\n",
       "      <td>0.062809</td>\n",
       "      <td>...</td>\n",
       "      <td>0.063689</td>\n",
       "      <td>0.072474</td>\n",
       "      <td>0.038251</td>\n",
       "      <td>0.009536</td>\n",
       "      <td>0.063812</td>\n",
       "      <td>0.090909</td>\n",
       "      <td>0.010512</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.001321</td>\n",
       "      <td>0.066126</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>693</td>\n",
       "      <td>NT12590</td>\n",
       "      <td>0.010153</td>\n",
       "      <td>0.015589</td>\n",
       "      <td>0.009506</td>\n",
       "      <td>0.000225</td>\n",
       "      <td>0.058593</td>\n",
       "      <td>0.009662</td>\n",
       "      <td>0.009038</td>\n",
       "      <td>0.079820</td>\n",
       "      <td>0.058861</td>\n",
       "      <td>...</td>\n",
       "      <td>0.058641</td>\n",
       "      <td>0.064128</td>\n",
       "      <td>0.038251</td>\n",
       "      <td>0.009152</td>\n",
       "      <td>0.058761</td>\n",
       "      <td>0.047619</td>\n",
       "      <td>0.009823</td>\n",
       "      <td>0.001321</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.057711</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>694</td>\n",
       "      <td>NT12400</td>\n",
       "      <td>0.059328</td>\n",
       "      <td>0.059486</td>\n",
       "      <td>0.058116</td>\n",
       "      <td>0.056974</td>\n",
       "      <td>0.020034</td>\n",
       "      <td>0.062489</td>\n",
       "      <td>0.056134</td>\n",
       "      <td>0.048151</td>\n",
       "      <td>0.010244</td>\n",
       "      <td>...</td>\n",
       "      <td>0.022776</td>\n",
       "      <td>0.040565</td>\n",
       "      <td>0.005319</td>\n",
       "      <td>0.057485</td>\n",
       "      <td>0.022792</td>\n",
       "      <td>0.105263</td>\n",
       "      <td>0.057306</td>\n",
       "      <td>0.066126</td>\n",
       "      <td>0.057711</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>695 rows × 696 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         0         1         2         3         4         5         6    \\\n",
       "0    NT12219  0.000000  0.016369  0.009429  0.010343  0.058500  0.011262   \n",
       "1    NT12588  0.016369  0.000000  0.014328  0.015217  0.056194  0.017083   \n",
       "2    NT12359  0.009429  0.014328  0.000000  0.009740  0.053117  0.010871   \n",
       "3    NT12611  0.010343  0.015217  0.009740  0.000000  0.055359  0.009888   \n",
       "4    NT12856  0.058500  0.056194  0.053117  0.055359  0.000000  0.061167   \n",
       "..       ...       ...       ...       ...       ...       ...       ...   \n",
       "690  NT12184  0.125000  0.083333  0.142857  0.095238  0.117647  0.050000   \n",
       "691  NT12612  0.010252  0.016821  0.010199  0.010101  0.059919  0.011269   \n",
       "692  NT12452  0.010522  0.017205  0.010225  0.001640  0.062591  0.010555   \n",
       "693  NT12590  0.010153  0.015589  0.009506  0.000225  0.058593  0.009662   \n",
       "694  NT12400  0.059328  0.059486  0.058116  0.056974  0.020034  0.062489   \n",
       "\n",
       "          7         8         9    ...       686       687       688  \\\n",
       "0    0.000680  0.070663  0.058651  ...  0.057861  0.055464  0.041451   \n",
       "1    0.013521  0.076053  0.056446  ...  0.056339  0.065170  0.038674   \n",
       "2    0.009328  0.081058  0.058294  ...  0.053201  0.063365  0.032258   \n",
       "3    0.009260  0.080031  0.057796  ...  0.055181  0.062853  0.038462   \n",
       "4    0.053473  0.060109  0.021639  ...  0.000000  0.049782  0.003058   \n",
       "..        ...       ...       ...  ...       ...       ...       ...   \n",
       "690  0.083333  0.066667  0.111111  ...  0.117647  0.125000  0.000000   \n",
       "691  0.009738  0.070707  0.059038  ...  0.059298  0.052690  0.032967   \n",
       "692  0.009252  0.089769  0.062809  ...  0.063689  0.072474  0.038251   \n",
       "693  0.009038  0.079820  0.058861  ...  0.058641  0.064128  0.038251   \n",
       "694  0.056134  0.048151  0.010244  ...  0.022776  0.040565  0.005319   \n",
       "\n",
       "          689       690       691       692       693       694       695  \n",
       "0    0.000519  0.057934  0.125000  0.010252  0.010522  0.010153  0.059328  \n",
       "1    0.013868  0.056473  0.083333  0.016821  0.017205  0.015589  0.059486  \n",
       "2    0.009290  0.053271  0.142857  0.010199  0.010225  0.009506  0.058116  \n",
       "3    0.009600  0.055303  0.095238  0.010101  0.001640  0.000225  0.056974  \n",
       "4    0.053902  0.000000  0.117647  0.059919  0.062591  0.058593  0.020034  \n",
       "..        ...       ...       ...       ...       ...       ...       ...  \n",
       "690  0.090909  0.117647  0.000000  0.150000  0.090909  0.047619  0.105263  \n",
       "691  0.010199  0.059373  0.150000  0.000000  0.010512  0.009823  0.057306  \n",
       "692  0.009536  0.063812  0.090909  0.010512  0.000000  0.001321  0.066126  \n",
       "693  0.009152  0.058761  0.047619  0.009823  0.001321  0.000000  0.057711  \n",
       "694  0.057485  0.022792  0.105263  0.057306  0.066126  0.057711  0.000000  \n",
       "\n",
       "[695 rows x 696 columns]"
      ]
     },
     "execution_count": 93,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "distance_matrix"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
